flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/DefaultDeclarativeSlotPool.java

```diff
@@ -127,7 +155,28 @@ public void increaseResourceRequirementsBy(ResourceCounter increment) {
         }
         totalResourceRequirements = totalResourceRequirements.add(increment);
 
+        if (slotRequestMaxInterval == null) {
+            declareResourceRequirements();
+            return;
+        }
+
+        Preconditions.checkNotNull(componentMainThreadExecutor);
+        if (slotRequestMaxIntervalTimeoutCheckFuture != null) {
+            slotRequestMaxIntervalTimeoutCheckFuture.cancel(true);
```

Comment:
In current implementation, the resource requirement will be sent if there is no further requriement change within the `slotRequestMaxInterval`. Another choice is to declare requirement at most once every `slotRequestMaxInterval`. TBH I'm not sure which one would be better. But I think that deserves a discussion.

Reply:
Thank you very much for the comments. 
Both are acceptable to me.
But I can't make a good evaluation on its in my limited read. Would you mind explaining the advantages of another approach?
Thank you very much~
CC @1996fanrui

Reply:
In current approach, we will minimize the rpc calls. The other approach may benefit the e2e latency.